hypothesis-python/tests/nocover/test_precise_shrinking.py

```diff
@@ -0,0 +1,261 @@
+    try:
+        buf = runner.cached_test_function(initial_buffer)
+        assert buf.status == Status.INTERESTING
+        assert buf.buffer == initial_buffer
+        assert runner.interesting_examples
+        runner.shrink_interesting_examples()
+    except RunIsComplete:
+        assert runner.exit_reason in (ExitReason.finished, ExitReason.max_shrinks)
+    (result,) = runner.interesting_examples.values()
+
+    data = ConjectureData.for_buffer(result.buffer)
+    result_value = safe_draw(data, strategy)
+    data.freeze()
+    return data.as_result(), result_value
+
+
+common_strategies_with_types = [
+    (type(None), st.none()),
+    (bool, st.booleans()),
+    (bytes, st.binary()),
+    (str, st.text()),
+    (int, st.integers()),
+]
+
+common_strategies = [v for _, v in common_strategies_with_types]
+
+
+@lru_cache
+def minimal_buffer_for_strategy(s):
+    return precisely_shrink(s, end_marker=st.none())[0].buffer
+
+
+def test_strategy_list_is_in_sorted_order():
+    assert common_strategies == sorted(
+        common_strategies, key=lambda s: shortlex(minimal_buffer_for_strategy(s))
+    )
+
+
+@pytest.mark.parametrize("typ,strat", common_strategies_with_types)
+@pytest.mark.parametrize("require_truthy", [False, True])
+def test_can_precisely_shrink_values(typ, strat, require_truthy):
+    if typ is type(None) and require_truthy:
+        pytest.skip("None is falsey")
+    if require_truthy:
+        cond = bool
+    else:
+        cond = lambda x: True
+    result, shrunk = precisely_shrink(strat, is_interesting=cond)
+    assert shrunk == find(strat, cond)
+
+
+alternatives = [
+    comb
+    for n in (2, 3, 4)
+    for comb in itertools.combinations(common_strategies_with_types, n)
+]
+
+indexed_alternatives = [
+    (i, j, a) for a in alternatives for i, j in itertools.combinations(range(len(a)), 2)
+]
+
+
+@pytest.mark.parametrize("i,j,a", indexed_alternatives)
+@pytest.mark.parametrize("seed", [0, 4389048901])
+def test_can_precisely_shrink_alternatives(i, j, a, seed):
+    types = [u for u, _ in a]
+    combined_strategy = st.one_of(*[v for _, v in a])
+
+    result, value = precisely_shrink(
+        combined_strategy,
+        initial_condition=lambda x: isinstance(x, types[j]),
+        is_interesting=lambda x: not any(isinstance(x, types[k]) for k in range(i)),
+    )
+    assert isinstance(value, types[i])
+
+
+@pytest.mark.parametrize(
+    "a", list(itertools.combinations(common_strategies_with_types, 3))
+)
+@pytest.mark.parametrize("seed", [0, 4389048901])
+def test_precise_shrink_with_blocker(a, seed):
+    # We're reordering this so that there is a "blocking" unusually large
+    # strategy in the middle.
+    x, y, z = a
+    a = (x, z, y)
+
+    types = [u for u, _ in a]
+    combined_strategy = st.one_of(*[v for _, v in a])
+
+    result, value = precisely_shrink(
+        combined_strategy,
+        initial_condition=lambda x: isinstance(x, types[2]),
+        is_interesting=lambda x: True,
```

Comment:
```suggestion
        is_interesting=lambda x: True,
        seed=seed,
```
ditto

Reply:
This seems like another mysterious nondiff to me. What am I missing?